tools/src/main/scala/scala/scalanative/codegen/llvm/CodeGen.scala

```diff
@@ -68,12 +68,41 @@ object CodeGen {
       val env = assembly.map(defn => defn.name -> defn).toMap
       val workDir = VirtualDirectory.real(config.workDir)
 
+      /**
+       *    - \C:a\b\c.txt -> a\b\c.txt
+       *    - C:a\b\c.txt -> a\b\c.txt
+       *    - p:\a\b\c.txt -> a\b\c.txt
+       *    - /a/b/c.txt -> a/b/c.txt
+       */
+      def dropPrefix(fileName: String): String = {
+        val driveLetter = s"^[A-Za-z]:".r
+        val supportDriveLetter =
+          build.Platform.isWindows || build.Platform.isCygwin || build.Platform.isMsys
+        // using "/" instead of `File.separator`
+        // we get "/C:a/b/c.txt" even in windows for some reasons.
+        val noDriveLetter =
+          driveLetter.findFirstIn(fileName.stripPrefix("/")) match {
+            case Some(prefix) if supportDriveLetter =>
+              fileName.stripPrefix(prefix)
+            case _ => fileName
+          }
+        noDriveLetter.stripPrefix("/")
+      }
+
+      def sourceDirOf(pos: Position): String = {
+        lazy val empty = "__empty"
```

Comment:
This lazy val should be passed as constant directly, it would be handle by the JVM runtime as literal and would not introduce any overhead

Reply:
Otherwise we can move it outside the methods and make it `private final val EmptyPath = "__empty"` to achive the same result